CONTRIBUTING.md

```diff
@@ -0,0 +1,114 @@
+# Contribution Guide
+
+We welcome and encourage community contributions to `rack`.
+
+Please familiarize yourself with the Contribution Guidelines and Project Roadmap before contributing.
+
+There are many ways to help `rack` besides contributing code:
+
+ - Fix bugs or file issues
+ - Improve the [documentation](https://github.com/jrperritt/rack/tree/master/docs)
+
+## Contributing Code
+
+Unless you are fixing a known bug, we **strongly** recommend discussing it with the core team via a GitHub issue or on [Freenode IRC](irc://chat.freenode.net/rackspace) before getting started to ensure your work is consistent with `rack`'s roadmap and architecture.
+
+All contributions are made via pull request. Note that **all patches from all contributors get reviewed**. After a pull request is made other contributors will offer feedback, and if the patch passes review a maintainer will accept it with a comment. When pull requests fail testing, authors are expected to update their pull requests to address the failures until the tests pass and the pull request merges successfully.
+
+At least one review from a maintainer is required for all patches (even patches from maintainers).
+
+Reviewers should leave a `LGTM` or `:+1:` comment once they are satisfied with the patch. If the patch was submitted by a maintainer with write access, the pull request should be merged by the submitter after review.
+
+## Code Style
+
+Please follow these guidelines when formatting source code:
+
+* Go code should match the output of `gofmt -s`
+* Shell scripts should adhere to the [Google Shell Style guide](https://google-styleguide.googlecode.com/svn/trunk/shell.xml)
```

Comment:
Our current `prep-travis-release.sh` needs to be adapted to this, particularly error messages on stderr (return codes are right though).